controllers/eventsource/resource.go

```diff
@@ -409,3 +414,36 @@ func envFromSecretsOrConfigMaps(source interface{}, t reflect.Type) []corev1.Env
 	}
 	return result
 }
+
+func buildGcpPubsubCredentialsVolumes(pubsubs map[string]v1alpha1.PubSubEventSource) ([]corev1.Volume, []corev1.VolumeMount) {
```

Comment:
Don't need this. All the `secretKeySelector` and `configMapKeySelector` will be automatically added to `envFrom`, as a result you can get the secret from ENV. 

See `func envFromSources`, and refer to other event sources with credential secrets.

Reply:
So is it better to inject secret via environmental variable, rather than file?

Reply:
It is not better or worse, but easier.

Also, we have more than 20 types of event sources, about half of them have secrets need to inject, if use volume for them, we should also use an unified way instead of doing it one by one.

Reply:
Ah I see! Your point makes sence.
I'll switch to use env var so that it'll be aligned better with other even source.